app/classifier/tasks/shortcut-editor.jsx

```diff
@@ -0,0 +1,136 @@
+import React from 'react';
+import AutoSave from '../../components/auto-save';
+import handleInputChange from '../../lib/handle-input-change';
+import Shortcut from './shortcut';
+
+export default class ShortcutEditor extends React.Component {
+  constructor(props) {
+    super(props);
+    this.addAnswer = this.addAnswer.bind(this);
+    this.toggleShortcut = this.toggleShortcut.bind(this);
+  }
+
+  toggleShortcut(e) {
+    let nextTaskID;
+
+    if (e.target.checked) {
+      const taskCount = Object.keys(this.props.workflow.tasks).length;
+
+      let taskIDNumber = -1;
+
+      while (!((nextTaskID != null) && !(nextTaskID in this.props.workflow.tasks))) {
+        taskIDNumber += 1;
+        nextTaskID = `T${taskCount + taskIDNumber}`;
+      }
+
+      const changes = {};
+      changes[`tasks.${nextTaskID}`] = Shortcut.getDefaultTask(this.props.task.question);
+
+      this.props.task.unlinkedTask = nextTaskID;
+      this.props.workflow.update(changes);
+      this.addAnswer();
+    } else {
+      delete this.props.workflow.tasks[this.props.task.unlinkedTask];
+      delete this.props.task.unlinkedTask;
+      this.props.workflow.update('tasks');
+    }
+  }
+
+  addAnswer() {
+    this.props.workflow.tasks[this.props.task.unlinkedTask].answers.push({ label: 'Nothing Here' });
+    this.props.workflow.update('tasks');
+  }
+
+  removeChoice(index) {
+    if (this.props.task) {
+      const answers = this.props.workflow.tasks[this.props.task.unlinkedTask].answers;
+      answers.splice(index, 1);
+      if (!answers.length) {
+        delete this.props.workflow.tasks[this.props.task.unlinkedTask];
+        delete this.props.task.unlinkedTask;
+      }
+    }
+    this.props.workflow.update('tasks');
+  }
+
+  render() {
+    let shortcuts;
+    if (this.props.task) {
+      shortcuts = this.props.workflow.tasks[this.props.task.unlinkedTask];
+    }
+    const handleChange = handleInputChange.bind(this.props.workflow);
+    const children = React.Children.map(this.props.children, (child) => {
+      return React.cloneElement(child);
+    });
+
+    return (
+      <div>
+
+        {children}
+        <hr />
+
+        <label htmlFor="shortcut" title="Shortcut Options to End Classification">
+          <AutoSave resource={this.props.workflow}>
+            <span className="form-label">Shortcut Option</span>{' '}
+            <input type="checkbox" checked={shortcuts != null} onChange={this.toggleShortcut} />
```

Comment:
Use strict equality whenever possible in javascript. Loose equality, `==` and `!=` can have unexpected results because it does type conversion: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Equality_comparisons_and_sameness